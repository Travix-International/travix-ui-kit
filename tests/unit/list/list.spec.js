import * as enzyme from 'enzyme';
import * as enzymeToJson from 'enzyme-to-json';
import React from 'react';
import List from '../../../components/list/list';

describe('List', () => {
  describe('#render()', () => {
    const shallow = enzyme.shallow;
    const shallowToJson = enzymeToJson.shallowToJson;

    it('should not modify mods', () => {
      const mods = ['test'];
      const wrapper = shallow(
        <List
          align="vertical"
          items={['London', 'Amsterdam', 'Madrid']}
          mods={mods}
        />
      );

      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods', () => {
      const wrapper = shallow(
        <List align="vertical" items={['London', 'Amsterdam', 'Madrid']} />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods', () => {
      const wrapper = shallow(
        <List align="horizontal" items={['London', 'Amsterdam', 'Madrid']} />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods', () => {
      const wrapper = shallow(
        <List align="vertical" hideBullets items={['London', 'Amsterdam', 'Madrid']} />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
