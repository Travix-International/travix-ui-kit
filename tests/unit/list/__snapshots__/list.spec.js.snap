// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List #render() should not modify mods 1`] = `
<ul
  className="ui-list ui-list_test ui-list_align_vertical"
>
  <li
    className="ui-list__item"
  >
    London
  </li>
  <li
    className="ui-list__item"
  >
    Amsterdam
  </li>
  <li
    className="ui-list__item"
  >
    Madrid
  </li>
</ul>
`;

exports[`List #render() should return base class with mods for strings as class and string as mods 1`] = `
<ul
  className="ui-list ui-list_align_vertical"
>
  <li
    className="ui-list__item"
  >
    London
  </li>
  <li
    className="ui-list__item"
  >
    Amsterdam
  </li>
  <li
    className="ui-list__item"
  >
    Madrid
  </li>
</ul>
`;

exports[`List #render() should return base class with mods for strings as class and string as mods 2`] = `
<ul
  className="ui-list ui-list_align_horizontal"
>
  <li
    className="ui-list__item"
  >
    London
  </li>
  <li
    className="ui-list__item"
  >
    Amsterdam
  </li>
  <li
    className="ui-list__item"
  >
    Madrid
  </li>
</ul>
`;

exports[`List #render() should return base class with mods for strings as class and string as mods 3`] = `
<ul
  className="ui-list ui-list_align_vertical ui-list_no-bullets"
>
  <li
    className="ui-list__item"
  >
    London
  </li>
  <li
    className="ui-list__item"
  >
    Amsterdam
  </li>
  <li
    className="ui-list__item"
  >
    Madrid
  </li>
</ul>
`;
